### PR TITLE
fix: add chainName to LoginStrategyOptions and set on cache client co…

### PR DESCRIPTION
**Pull request by whitneypurdum**
- add chainName to LoginStrategyOptions interface
- use chainName in cacheServerClient to identify chain when creating did

### CODE REVIEW

File: lib/LoginStrategy.ts [hunk 1]
```diff
@@ -302,7 +303,12 @@ export class LoginStrategy extends BaseStrategy {
         null
       );
     }
-    const userDid = this.didUnification(`did:ethr:${payload.address}`);
+    let userDid;
+    if ((await this.getChainName(this.provider)) === 'ewc') {
```
Comment (jrhender) [inline lib/LoginStrategy.ts:307]: Better to rely on the chainId of the SIWE message I think. It is more intuitive as it allows the DID to be generated solely from the message.
```suggestion
    if (payload.chainId === EWC_CHAIN_ID) {
```
If above is used, `getChainName` can also be removed

**Comment by jrhender** [comment]
@nichonien how hard would it be to add a test for this case?

**Comment by jrhender** [comment]
@nichonien I think a test would involve adjusting the SIWE payload to have the EWC chain ID https://github.com/energywebfoundation/passport-did-auth/blob/develop/test/LoginStrategy.siwetoken.test.ts#LL40C14-L40

**Review by jrhender — APPROVED** [review]
@nichonien perhaps we can merge this as is and then add a test in a separate PR? How does that sound?

**Comment by nichonien** [comment]
> @nichonien I think a test would involve adjusting the SIWE payload to have the EWC chain ID https://github.com/energywebfoundation/passport-did-auth/blob/develop/test/LoginStrategy.siwetoken.test.ts#LL40C14-L40

Yes, I am trying to obtain a `siwe` signature with the `message` in order to add it to tests.

**Comment by nichonien** [comment]
> @nichonien perhaps we can merge this as is and then add a test in a separate PR? How does that sound?

Yeah, sounds good. We can merge this PR now and I'll create another PR with tests once I get the required sample payload with `ewc` chainID

**Pull request merged**